### PR TITLE
[Documentation] add enhancement.yml

### LINKED ISSUE #7241: [Feature] Add issue type `Enhancement`
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

I think that it is essential to add issue type `Enhancement`.

`Enhancement` is different from `Feature request` which is for users, we should add a type for developer/contributor of doris.

### Use case

None

### Related issues

None

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jackwener**
## Proposed changes

Fix #7241

Add enhancement type of issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jackwener** [comment]
I forgot to edit information.

File: .github/ISSUE_TEMPLATE/enhancement.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+name: Doris enhancement
+description: Add an enhanment for Apache Doris
+title: "[Enhancement] "
+labels: [ "kind/enhancement" ]
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thank you very much for your good enhancement for Doris.
```
Comment (morningman) [inline .github/ISSUE_TEMPLATE/enhancement.yml:26]: Please add
`Please help sync to the [Dev Mail List](mailto:dev@doris.apache.org). You will get FASTER help！[How to subscribe](mailto:dev-subscribe@doris.apache.org)`

Comment (morningman) [reply .github/ISSUE_TEMPLATE/enhancement.yml:26]: you can refer to `.github/ISSUE_TEMPLATE/feature-request.yml`

Comment (jackwener) [reply .github/ISSUE_TEMPLATE/enhancement.yml:26]: fixed

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**